### ISSUE TITLE
There is an error of your codes? Maybe

### ISSUE THREAD

**Issue by wxqlab** [HeliosZhao/NCDSS#7]
I have a bit of a doubt about this code of base_train.py. I think the location of some codes should be reversed, right?

    for name, param in model.named_parameters():
        for ft_layer in p['ft_layer']:
            if name.startswith(ft_layer):

                param.requires_grad = True
                if name.startswith('decoder'):
                    print('Add {} in decoder params'.format(name))
                    decoder_params.append(param)
                else:
                    print('Add {} in classifier params'.format(name))
                    classifier_params.append(param)
                
                break
            else:
                param.requires_grad = False

Could you please tell the answer？thank you very much!!!

**Comment by HeliosZhao** [comment]
Sorry for the confusion.

The code in `base_train.py` is correct, but the name is not very appropriate. In `p["ft_layer"]`, there are three layers `['backbone.layer3', 'backbone.layer4', 'decoder']`, the backbone layers should use a small learning rate while the decoder layers should use a large learning rate.

In [optimizer](https://github.com/HeliosZhao/NCDSS/blob/0116e9fb9953dd5b2e29c58357946e4bfef01c8a/base_train.py#L110), we set the `decoder_params` with small learning rate while `classifier_params` with large learning rate. So the `backbone.layer3', 'backbone.layer4'` are put into the `decoder_params`.

**Comment by wxqlab** [comment]
@HeliosZhao 
Thank you very much for your reply!
